organize/models.py

```diff
@@ -28,20 +35,144 @@ class EventApplication(models.Model):
+    status = models.CharField(
+        choices=APPLICATION_STATUS,
+        default=NEW,
+        max_length=10,
+    )
+    status_changed_at = models.DateTimeField(null=True, blank=True)
+
+    comment = models.TextField(null=True, blank=True)
 
     class Meta:
         permissions = (
             ("can_accept_organize_application",
              "Can accept Organize Applications"),
         )
 
+    def __str__(self):
+        return "{}, {} ({})".format(self.city, self.country, self.get_status_display())
+
+    def create_event(self):
+        """ Creates event based on the data from the EventApplication.
+        """
+        name = 'Django Girls {}'.format(self.city)
+        email = '{}@djangogirls.org'.format(self.website_slug)
+
+        event = Event.objects.create(
+            date=self.date,
+            city=self.city,
+            country=self.country,
+            latlng=self.latlng,
+            page_url=self.website_slug,
+            name=name,
+            page_title=name,
+            email=email,
+        )
+
+        # populate content & menu from the default event
+        event.add_default_content()
+        event.add_default_menu()
+
+        return event
+
+    @transaction.atomic
+    def deploy(self):
+        """ Deploy Event based on the current EventApplication
+            - change status to DEPLOYED
+            - creates or copies event
+            - add/remove organizers
+            - send email about deployment
+        """
+        if self.status == DEPLOYED:  # we don't want to deploy twice
+            return
+
+        self.change_status_to(DEPLOYED)
+
+        # TODO: we should recognize here if we should create a new event,
+        # copy old one or copy old and change organizaers.
+        event = self.create_event()
+
+        # TODO: use method created in separate branch to create gmail acconut
+        # and get password from it.
+        password = "FAKE_PASS"
+
+        # TODO: remove organizers, who are no longer in org team if cloned
+        event.create_organizers(
+            team=self.coorganizers.all(),
+            email_password=password
+        )
+        self.send_event_deployed_email(event, email_password=password)
+
+    def clean(self):
+        if self.status == ON_HOLD and not self.comment:
+            raise ValidationError({
+                'comment': 'This field is required.'
+            })
+
+    def get_all_recipients(self):
+        """
+        Returns a list of emails to all organizers in that application
+        """
+        emails = [coorganizer.email for coorganizer in self.coorganizers.all()]
+        emails.append(self.main_organizer_email)
+        return emails
+
+    def change_status_to(self, status):
+        """ Changes status to the status provided
+            - sets proper status_changed_at datetime
+        """
+        self.status = REJECTED
+        self.status_changed_at = timezone.now()
+        self.save()
```

Comment:
Using `self.save(update_fields=['status', 'status_changed_at'])` here would be a bit nicer but it's not a strictly equivalent change. I don't know if some parts of the code rely on all fields (not just status) to be saved.